com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/utils/FileSystemUtils.java

```diff
@@ -108,4 +114,143 @@ public static String decodeURI(String uri) {
       return uri;
     }
   }
+
+  /**
+   * This method return a {@link Path} representation of a bunch of {@link String} given as input or
+   * null if is not possible get the path. After each pah a FS separator is added automatically
+   *
+   * @param basePath the string that represent the base path
+   * @param more additional String used to compose the path
+   * @return the {@link Path} that contains all the given segment, null if the path doesn't exists.
+   */
+  public Path getPath(@Nonnull String basePath, @Nonnull String... more) {
+    return Paths.get(
+        basePath,
+        filesystemSeparator(),
+        Arrays.stream(more)
+            .map(path -> path + filesystemSeparator())
+            .collect(Collectors.joining()));
+  }
+
+  /**
+   * This method create a folder from a given path
+   *
+   * @param path in input that will be used to create the folder
+   */
+  public void createFolder(Path path) {
+    try {
+      Files.createDirectory(path);
+    } catch (IOException e) {
+      log.error(e.getMessage());
+    }
+  }
+
+  /**
+   * Create a file on filesystem from a given {@link Path}
+   *
+   * @param path target path that will be used to create the file
+   */
+  public void createFile(Path path) {
+    if (!path.toFile().exists()) {
```

Comment:
You may use `Files#exists` to match the Law of Demetra

Reply:
Nice reference to the Demetra's Law! 
Btw I followed this suggestion from sonarcloud: https://rules.sonarsource.com/java/tag/performance/RSPEC-3725

Reply:
Oh, I didn't know that, thanks!